lms/djangoapps/verify_student/api.py

```diff
@@ -33,3 +40,63 @@ def send_approval_email(attempt):
     else:
         email_context = {'user': attempt.user, 'expiration_datetime': expiration_datetime.strftime("%m/%d/%Y")}
         send_verification_approved_email(context=email_context)
+
+
+def create_verification_attempt(user, name, status, expiration_datetime=None):
+    """
+    Create a verification attempt.
+
+    This method is intended to be used by IDV implementation plugins to create VerificationAttempt instances.
+
+    Args:
+        user (User): the user (usually a learner) performing the verification attempt
+        name (string): the name being ID verified
+        status (string): the initial status of the verification attempt
+        expiration_datetime (datetime, optional): When the verification attempt expires. Defaults to None.
+
+    Returns:
+        id (int): The id of the created VerificationAttempt instance
+    """
+    verification_attempt = VerificationAttempt.objects.create(
+        user=user,
+        name=name,
+        status=status,
+        expiration_datetime=expiration_datetime,
+    )
+
+    return verification_attempt.id
+
+
+def update_verification_attempt_status(attempt_id, status):
+    """
+    Update the status of a verification attempt.
+
+    This method is intended to be used by IDV implementation plugins to update VerificationAttempt instances.
+
+    Arguments:
+        * id (str): the verification attempt id of the attempt to update
+        * status (str): the new status
+
+    Returns:
+        * None
+    """
+    status_list = [attr for attr in dir(VerificationAttemptStatus) if not attr.startswith('__')]
```

Comment:
My concern here is that this is brittle. If any other attributes are added to the class, they'll be treated as statuses - even methods!

I see three alternatives.

1. We could enumerate the list of valid statuses here.
2. We could add an attribute like `all_statuses` to the `VerificationAttemptStatus` class.
3. We could add an `is_valid_status` to the `VerificationAttemptStatus` class that `update_verification_attempt_status` class calls.

I'm partial to #3, because then the status class is responsible for defining what a valid status is, not the API function.

What do you think?

Reply:
In retrospect, trying to save a non-valid status (i.e. something that isn't in the `choices` list on the model) will raise a `ValidationError`, so Django handles this for us out of the box. You could consider keeping the default Django behavior instead of explicitly checking the status. My mistake.

Reply:
Added a comment below with how I decided to handle this: https://github.com/openedx/edx-platform/pull/35338#discussion_r1735144386